x-pack/platform/plugins/shared/fleet/server/routes/epm/handlers.ts

```diff
@@ -664,3 +667,44 @@ const soToInstallationInfo = (pkg: PackageListItem | PackageInfo) => {
   }
   return pkg;
 };
+
+export const rollbackPackageHandler: FleetRequestHandler<
+  TypeOf<typeof RollbackPackageRequestSchema.params>
+> = async (context, request, response) => {
+  const { pkgName } = request.params;
+  const coreContext = await context.core;
+  const esClient = coreContext.elasticsearch.client.asInternalUser;
+  const fleetContext = await context.fleet;
+  // Need a less restrictive client than fleetContext.internalSoClient for SO operations in multiple spaces.
+  const savedObjectsClient = new SavedObjectsClient(
+    appContextService.getSavedObjects().createInternalRepository()
+  );
+  const spaceId = fleetContext.spaceId;
+  const allSpaces = await fleetContext.getAllSpaces();
+  const logger = appContextService.getLogger();
+  try {
+    const body: RollbackPackageResponse = await rollbackInstallation({
+      esClient,
+      savedObjectsClient,
+      pkgName,
+      spaceId,
+      spaceIds: allSpaces.map((space) => space.id),
+    });
+    return response.ok({ body });
+  } catch (error) {
+    const message = `Failed to rollback package ${pkgName}: ${error.message}`;
+    if (error instanceof PackageRollbackError) {
+      logger.warn(message);
+      return response.customError({
+        statusCode: 400,
```

Comment:
`PackageRollbackError` should extend `FleetError`, so HTTP 400 will be returned without the extra  `response.customError`, see https://github.com/elastic/kibana/blob/main/x-pack/platform/plugins/shared/fleet/server/errors/handlers.ts#L67